packages/libs/eda/src/lib/core/components/computations/plugins/differentialabundance.tsx

```diff
@@ -0,0 +1,273 @@
+      analysisState,
+      computation,
+      visualizationId
+    );
+
+  const collectionVarItems = useMemo(() => {
+    return collections.map((collectionVar) => ({
+      value: {
+        variableId: collectionVar.id,
+        entityId: collectionVar.entityId,
+      },
+      display:
+        collectionVar.entityDisplayName + ' > ' + collectionVar.displayName,
+    }));
+  }, [collections]);
+
+  const selectedCollectionVar = useMemo(() => {
+    if (configuration && 'collectionVariable' in configuration) {
+      const selectedItem = collectionVarItems.find((item) =>
+        isEqual(item.value, {
+          variableId: configuration.collectionVariable.variableId,
+          entityId: configuration.collectionVariable.entityId,
+        })
+      );
+      return selectedItem;
+    }
+  }, [collectionVarItems, configuration]);
+
+  const selectedComparatorVariable = useMemo(() => {
+    if (
+      configuration &&
+      configuration.comparator &&
+      'variable' in configuration.comparator
+    ) {
+      return findEntityAndVariable(configuration.comparator.variable);
+    }
+  }, [configuration]);
+
+  return (
+    <ComputationStepContainer
+      computationStepInfo={{
+        stepNumber: 1,
+        stepTitle: `Configure ${computationAppOverview.displayName}`,
+      }}
+    >
+      <div style={sharedConfigCssStyles}>
+        <div
+          style={{
+            display: 'flex',
+            gap: '1em',
+            justifyItems: 'start',
+            alignItems: 'center',
+          }}
+        >
+          <div style={{ justifySelf: 'end', fontWeight: 500 }}>Data</div>
+          <SingleSelect
+            value={
+              selectedCollectionVar
+                ? selectedCollectionVar.value
+                : 'Select the data'
+            }
+            buttonDisplayContent={
+              selectedCollectionVar
+                ? selectedCollectionVar.display
+                : 'Select the data'
+            }
+            items={collectionVarItems}
+            onSelect={partial(changeConfigHandler, 'collectionVariable')}
+          />
+          <div style={{ justifySelf: 'end', fontWeight: 500 }}>
+            Comparator Variable
+          </div>
+          <VariableTreeDropdown
+            showClearSelectionButton={false}
+            scope="variableTree"
+            showMultiFilterDescendants
+            starredVariables={[]}
+            toggleStarredVariable={toggleStarredVariable}
+            entityId={configuration?.comparator?.variable?.entityId}
+            variableId={configuration?.comparator?.variable?.variableId}
+            variableLinkConfig={{
+              type: 'button',
+              onClick: (variable) => {
+                changeConfigHandler('comparator', {
+                  variable: variable as VariableDescriptor,
+                  groupA: configuration?.comparator?.groupA ?? null,
+                  groupB: configuration?.comparator?.groupB ?? null,
```

Comment:
I think we need to reset the group values when the comparator var changes. In my screenshot, I had selected group values based on `City, village or region` and then switched to `Delivery mode` and the newly irrelevant group values persisted.

![image](https://github.com/VEuPathDB/web-monorepo/assets/69446567/2ad2bccd-899c-4f25-a143-0a196663762b)

I also inadvertently made _**a lot**_ of computations when testing this work, so I think we may need to further investigate how/why/when we make a new computation. Frankly, I don't much recall how I derived that logic and would be happy to help investigate.

[Warning: dizzying screencast of all the computations!](https://github.com/VEuPathDB/web-monorepo/assets/69446567/0dc17925-87f5-4f67-b0cd-8d4bb33e6a61)

Reply:
> I think we need to reset the group values when the comparator var changes

Oh yes good catch! Done!

Reply:
Wow that's a lot of computations!! You've been busy!
I agree the how/when/why of making a new computation is worth a think. One part I'm torn over is if the variable values should contribute to making a new computation section or not. I'm planning to demo this at the mbio meeting this week and can ask dan what he thinks

Reply:
There is something very strange going on with this. Every time I open/close either the **Group A** or **Group B** selector, the number of visualizations doubles.

Reply:
> There is something very strange going on with this. Every time I open/close either the **Group A** or **Group B** selector, the number of visualizations doubles.

But this only happens if no new values are selected... very strange!

Reply:
Also, all of these vizs have the same id :face_with_head_bandage:

Reply:
I think there is a bug in this part of the code: https://github.com/VEuPathDB/web-monorepo/blob/8ba293186841a84e2406064b8e4fb08e7729c11f/packages/libs/eda/src/lib/core/components/computations/Utils.ts#L125-L151.

I think we need to check, earlier in this function, if the new compute config is indeed different than the existing compute config, and bail if it isn't.

We are getting here because the variable drop down will call the onchange handler, even if the value is the same. This causes the code to enter a path where it thinks it's moving a viz from one computation to another, but that is not what is happening.

Reply:
Try applying this diff:

```diff
diff --git a/packages/libs/eda/src/lib/core/components/computations/Utils.ts b/packages/libs/eda/src/lib/core/components/computations/Utils.ts
index 371b17be3d..9984f186ee 100644
--- a/packages/libs/eda/src/lib/core/components/computations/Utils.ts
+++ b/packages/libs/eda/src/lib/core/components/computations/Utils.ts
@@ -98,6 +98,9 @@ function handleConfigurationChanges<ConfigType>(
   url: string,
   history: RouterChildContext['router']['history']
 ) {
+
+  if (isEqual(computation.descriptor.configuration, updatedConfiguration)) return;
+  
   // when a config value changes:
   // 1. remove viz from current computation
   // 2. check if the newConfig exists
```